kubeconfig-creator/README.md

```diff
@@ -0,0 +1,155 @@
+
+## Parameters
+
+* **Name**: Name of the `cluster`.
+* **Type**: Type of the resource (_default:_
+  `cluster`)
+* **URL**: Address of the target cluster (_e.g.:_ 
+  `https://hostname:port`)
+* **Username**: Username for basic authentication to the cluster
+(_default:_ `""`)
+* **Password**: Password for basic authentication to the cluster
+(_default:_ `""`)
+* **Cadata**: Contains PEM-encoded certificate authority certificates
+(_default:_ `""`)
+* **Namespace**: Default namespace to use on unspecified requests
+(_default:_ `""`)
+* **Token**: Bearer token for authentication to the cluster
+(_default:_ `""`)
+* **Insecure**:  If true, skips the validity check for the server's certificate. 
+This will make your HTTPS connections insecure
+(_default:_ `false  `)
+
+
+## Usage
+
+This task uses a `shared workspace` with [`PVC`](https://kubernetes.io/docs/concepts/storage/persistent-volumes) to store the
+ `kubeconfig` in the `kubeconfigFile` directory. 
+Kubeconfig file is stored at `/workspace/kubeconfigFile/kubeconfig`.
+
+Task can be used with the other task in the pipeline to authenticate the cluster.
+In this example, pipeline has a task `kubeconfig-creator` that generates a 
+`kubeconfig file` for the cluster and the `test-task` uses that kubeconfig file and verifiy whether the
+application has an access to the cluster or not by using some `kubectl/oc` commands.
+
+```
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: kubeconfig-test-pipeline
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - name: kubeconfig-creator
+      taskRef:
+        name: kubeconfig-creator
+      workspaces:
+        - name: kubeconfigFile
+          workspace: shared-workspace
+      params:
+        - name: name
+          value: cluster-bot
+        - name: username
+          value: kubeadmin
+        - name: url
+          value: https://api.ci-ln-r68sydb-d5d6b.origin-ci-int-aws.dev.rhcloud.com:6443
+        - name: cadata
+          value: LS0tLS1....
+        - name: token
+          value: Xj95RM...
+    - name: authentication-test
+      taskRef:
+        name: authentication-test
+      workspaces:
+        - name: kubeconfigFile
+          workspace: shared-workspace
+      params:
+        - name: filename
+          value: kubeconfig
+      runAfter:
+        - kubeconfig-creator
+```
+`Test-task` uses shared-workspace to fetch the kubeconfig file form the
+kubeconfigFile directory and uses oc commands to check whether the `cluster` is configured or not.
+```
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: authentication-test
+spec:
+  params:
+    - name: filename
+      description: kubeconfig file name
+      type: string
+  workspaces:
+    - name: kubeconfigFile
+      readOnly: true
+  steps:
+    - name: get
+      image: quay.io/openshift/origin-cli:latest
+      script: |
+
+        export KUBECONFIG="$(workspaces.kubeconfigFile.path)/$(inputs.params.filename)"
```

Comment:
Here as well, `workspaces.kubeconfigFile.path` isn't the path to the kubeconfig file but to the directory that contains the kubeconfig file.  It's a little bit confusing.

Reply:
Okay, will update with appropriate word.